collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,230 @@
+# Extend Kyma CLI
+
+Kyma cli has been used to provision a cluster and install kyma into. We should look into extend the current functionality set such that we can provide functionality to access various features of kyma.
+
+The extension of functionality would enable people with no or less kubernetes knowledge can use kyma without any issue.
+
+## Security
+
+Kyma cli should support RBAC. This would be realised by downloading the requisite kubeconfig file. The admins should be able to use more infrastructure commands and developers should be able to use the kyma cluster with restricted rights.
+
+## Architecture
+
+  We should follow the modular approach with each component having its own library. This library can be used inside the kyma cli for acessing various components. The library contains all the implementation. The kyma cli should invoke this library. We can take inspiration from hydroform and implement it in similar way. Currently we have following modules in kyma:
+
+  1. Lambdas
```

Comment:
Will the project templating fall into the function module as well?

Reply:
yes. Since its a feature for functions.